src/Core/Checkout/Test/Customer/Repository/CustomerGroupValidatorTest.php

```diff
@@ -0,0 +1,86 @@
+<?php declare(strict_types=1);
+
+namespace Shopware\Core\Checkout\Test\Customer\Repository;
+
+use Doctrine\DBAL\Connection;
+use PHPUnit\Framework\TestCase;
+use Shopware\Core\Defaults;
+use Shopware\Core\Framework\Context;
+use Shopware\Core\Framework\DataAbstractionLayer\EntityRepositoryInterface;
+use Shopware\Core\Framework\DataAbstractionLayer\Search\Criteria;
+use Shopware\Core\Framework\DataAbstractionLayer\Write\WriteException;
+use Shopware\Core\Framework\Test\TestCaseBase\IntegrationTestBehaviour;
+use Shopware\Core\Framework\Uuid\Uuid;
+
+class CustomerGroupValidatorTest extends TestCase
+{
+    use IntegrationTestBehaviour;
+
+    /**
+     * @var EntityRepositoryInterface
+     */
+    private $customerGroupRepository;
+
+    /**
+     * @var EntityRepositoryInterface
+     */
+    private $customerRepository;
+
+    /**
+     * @var EntityRepositoryInterface
+     */
+    private $salesChannelRepository;
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        $this->customerGroupRepository = $this->getContainer()->get('customer_group.repository');
+        $this->customerRepository = $this->getContainer()->get('customer.repository');
+        $this->salesChannelRepository = $this->getContainer()->get('sales_channel.repository');
+
+        /** @var Connection $connection */
+        $connection = $this->getContainer()->get(Connection::class);
+        $connection->beginTransaction();
```

Comment:
this is already done in one of the traits.

Reply:
Nice